tools/toolchains/gcc.py

```diff
@@ -208,6 +223,12 @@ def get_compile_options(self, defines, includes, for_asm=False):
             opts = opts + self.get_config_option(config_header)
         return opts
 
+    def match_coverage_patterns(self, source):
+        for pattern in self.coverage_patterns:
```

Comment:
Document what this new function is doing.

Reply:
comments added